ui/src/app/components/sharded-dataflow-migration-details-form/sharded-dataflow-migration-details-form.component.ts

```diff
@@ -416,11 +427,45 @@ export class ShardedDataflowMigrationDetailsFormComponent implements OnInit {
     }
   }
 
+ 
   determineConnectionProfileInfoValidity(): boolean {
     let formValue = this.migrationProfileForm.value
     return formValue.host != null && formValue.port != null && formValue.user != null && formValue.password != null && formValue.newSourceProfile != null
   }
 
+  verifyTextJson() {
+    this.verifyingJson = true;
+    let formValue = this.migrationProfileForm.value;
+    let payload: IMigrationProfile
+    try {
+      payload = JSON.parse(formValue.textInput);
+    } catch (error) {
+      if (error instanceof SyntaxError) {
+        this.errorVerMsg = error.message
```

Comment:
Please validate what this looks like on the UI

Reply:
I have checked, for reference:
<img width="1205" alt="image" src="https://github.com/GoogleCloudPlatform/spanner-migration-tool/assets/35952883/961e2fdc-52b5-438c-a9db-dd279e2430ec">